lib/gitWorkflow.js

```diff
@@ -169,82 +184,68 @@ class GitWorkflow {
   }
 
   /**
-   * Applies back task modifications, and unstaged changes hidden in the stash.
-   * In case of a merge-conflict retry with 3-way merge.
+   * Remove unstaged changes to all partially staged files, to avoid tasks from seeing them
+   * This fails when a file has been renamed, since of the entries points to a file not on disk
+   * It should be safe to ignore errors here
    */
-  async applyModifications(ctx) {
-    const modifiedFiles = await this.execGit(['ls-files', '--modified'])
-    if (modifiedFiles) {
-      debug('Detected files modified by tasks:')
-      debug(modifiedFiles)
-      debug('Adding files to index...')
+  async hideUnstagedChanges() {
+    try {
       await Promise.all(
-        // stagedFileChunks includes staged files that lint-staged originally detected.
-        // Add only these files so any 3rd-party edits to other files won't be included in the commit.
-        this.stagedFileChunks.map(stagedFiles => this.execGit(['add', ...stagedFiles]))
+        this.partiallyStagedFiles.map(file => this.execGit(['checkout', '--force', '--', file]))
       )
-      debug('Done adding files to index!')
-    }
+    } catch {} // eslint-disable-line no-empty
```

Comment:
This empty catch is because in case of renames, the `FROM` file no longer exists on disk, and thus `git checkout FROM` fails. In order to handle this better, the `this.partiallyStagedFiles` should include data about the status of the file, but I didn't do that yet (lazy).

Reply:
Let's add the comment to code please

Reply:
I think I’ll just add the proper error handling.

Reply:
I added the error handler, but decided to leave it out of code coverage with a code comment.

Reply:
@okonet Are you sure you are ok with ignoring coverage from the try/catch here: https://github.com/okonet/lint-staged/blob/simplify/lib/gitWorkflow.js#L214

Should we just show lower-than-100% test coverage? It would be around 98.5 % without ignoring that try/catch.

I can't come up with an actual test for this, since it would mean that another git operation would have changed the list of edited files between the backup and this step (both before running the tasks). With manual testing, `git checkout -f -- <file>` will error when `<file>` does not match any git pathspec. The only case I know of is when renaming a file from A to B; `git checkout -f -- A` would then fail because git now tracks B.